src/qt/walletmodel.cpp

```diff
@@ -29,35 +30,36 @@
 #include <QDebug>
 #include <QMessageBox>
 #include <QSet>
-#include <QTimer>
 
 
-WalletModel::WalletModel(std::unique_ptr<interfaces::Wallet> wallet, interfaces::Node& node, const PlatformStyle *platformStyle, OptionsModel *_optionsModel, QObject *parent) :
-    QObject(parent), m_wallet(std::move(wallet)), m_node(node), optionsModel(_optionsModel), addressTableModel(nullptr),
+WalletModel::WalletModel(std::unique_ptr<interfaces::Wallet> wallet, interfaces::Node& node, const PlatformStyle *platformStyle, OptionsModel *_optionsModel, ClientModel *_clientModel, QObject *parent) :
+    QObject(parent), m_wallet(std::move(wallet)), m_node(node), optionsModel(_optionsModel), clientModel(_clientModel), addressTableModel(nullptr),
     transactionTableModel(nullptr),
     recentRequestsTableModel(nullptr),
     cachedEncryptionStatus(Unencrypted),
-    cachedNumBlocks(0)
+    pollingTimer(nullptr),
+    m_cached_last_update_tip(uint256())
 {
     fHaveWatchOnly = m_wallet->haveWatchOnly();
     addressTableModel = new AddressTableModel(this);
-    transactionTableModel = new TransactionTableModel(platformStyle, this);
+    transactionTableModel = new TransactionTableModel(platformStyle, clientModel, this);
     recentRequestsTableModel = new RecentRequestsTableModel(this);
 
     subscribeToCoreSignals();
 }
 
 WalletModel::~WalletModel()
 {
+    if (pollingTimer) pollingTimer->stop();
     unsubscribeFromCoreSignals();
 }
 
 void WalletModel::startPollBalance()
 {
     // This timer will be fired repeatedly to update the balance
-    QTimer* timer = new QTimer(this);
-    connect(timer, &QTimer::timeout, this, &WalletModel::pollBalanceChanged);
-    timer->start(MODEL_UPDATE_DELAY);
+    pollingTimer = new QTimer(this);
```

Comment:
These changes to the timer are unnecessary. The timer is stopped and destroyed automatically on `WalletModel` destruction.

Reply:
double checked and yep, cleaning them 👍 .